### ISSUE TITLE
Fix S3878 FN: When params are passed as array throught an attribute

### ISSUE THREAD

**Issue by splendid25** [SonarSource/sonar-dotnet#6977]
### Description

Rule S3878 (Remove this array creation and simply pass the elements) is not triggered on an attribute constructor.

### Repro steps
```csharp
class CustomAttribute : Attribute
{
    public CustomAttribute(params string[] values)
    {
        //...
    }
}

internal enum eCustom
{
    [Custom(new[] { "1", "2"})]
    A
}
```

### Expected behavior

Warning S3878 should appear on line :
```csharp
[Custom(new[] { "1", "2"})]
```
### Actual behavior

Warning are only raised when the constructor is called from a method, with `new` :
```csharp
new CustomAttribute(new[] { "" }); // S3878
```

### Related information

* .NET Framework 4.7.2
* Visual Studio 2019 16.11.25
* SonarLint for Visual Studio 2019 6.13.0.62767
* SonarAnalyzer.CSharp 8.53.0.62665
* Windows 10 22H2

**Comment by mary-georgiou-sonarsource** [comment]
Hello @splendid25,

thanks a lot for reporting this.
I confirm it is an FN.

I added the issue to our backlog.

**Comment by mary-georgiou-sonarsource** [comment]
This issue will be tracked in [NET-1254](https://sonarsource.atlassian.net/browse/NET-1254) (not publicly accessible)
